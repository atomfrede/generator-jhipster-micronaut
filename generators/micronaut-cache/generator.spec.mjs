import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'micronaut-cache';
const SUB_GENERATOR_NAMESPACE = `jhipster-micronaut:${SUB_GENERATOR}`;

describe('SubGenerator micronaut-cache of micronaut JHipster blueprint', () => {
  describe('run', () => {
    beforeAll(async function () {
      await helpers
        .run(SUB_GENERATOR_NAMESPACE)
        .withJHipsterConfig()
        .withOptions({
          ignoreNeedlesError: true,
        })
        .withJHipsterLookup()
        .withParentBlueprintLookup();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
