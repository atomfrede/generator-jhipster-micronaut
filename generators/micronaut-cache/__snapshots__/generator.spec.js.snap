// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator micronaut-cache of micronaut JHipster blueprint > run > should succeed 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/CacheConfiguration.java": {
    "stateCleared": "modified",
  },
}
`;
